extensions/amp-analytics/0.1/variables.js

```diff
@@ -126,78 +109,39 @@ export class VariableService {
     /** @private {!Window} */
     this.win_ = window;
 
-    /** @private {!Object<string, !Filter>} */
-    this.filters_ = map();
-
-    this.register_('default', new Filter(defaultFilter, /* allowNulls */ true));
-    this.register_('substr', new Filter(substrFilter));
-    this.register_('trim', new Filter(value => value.trim()));
-    this.register_('json', new Filter(value => JSON.stringify(value)));
-    this.register_('toLowerCase', new Filter(value => value.toLowerCase()));
-    this.register_('toUpperCase', new Filter(value => value.toUpperCase()));
-    this.register_('not', new Filter(value => String(!value)));
-    this.register_('base64', new Filter(value => btoa(value)));
-    this.register_('hash', new Filter(this.hashFilter_.bind(this)));
-    this.register_('if', new Filter(
-        (value, thenValue, elseValue) => value ? thenValue : elseValue, true));
+    /** @private {!Object<string, *>} */
+    this.filters_ = {};
```

Comment:
Question: `Filter` sounds like an `ExpansionOptions` to me. Does it have to be a variableService param? Sounds to me that `Filter` will remain the same `expansionOptions` in all cases, can we make it an individual service, or a const `ExpansionOptions`?

Reply:
we are calling `new ExpansionOptions()` several times from within `VariableService` since they don't change, shouldn't we keep them on `VariableService`?

Reply:
Yes... I can see that keeping only one instance in variable service makes sense. Maybe we should give it a better name lol...

Reply:
should we call everything macros?

Reply:
named everything macro(s)